web/src/views/FileBrowserView/PublishFileBrowser.vue

```diff
@@ -142,8 +154,27 @@ export default {
     },
     items(items) {
       if (items && !items.length) {
+        // If the API call returns no items, go back to the root (shouldn't normally happen)
         this.location = rootDirectory;
+        return;
       }
+
+      // Create the download link in itemDownloads for each item in items
+      this.itemDownloads = {};
+      const { identifier, version, location } = this;
+      const baseAssetPath = `${publishRest.defaults.baseURL}/dandisets`;
+
+      items.filter((x) => !x.folder).map(async (item) => {
+        const relativePath = `${location}${item.name}`;
+        const assetPath = `${baseAssetPath}/${identifier}/versions/${version}/assets/`;
```

Comment:
Organizationally, I would prefer keeping API URLs segregated in `rest.js` rather than mixing them into components. Publish API endpoints should each have a method on `publishRest` so components don't have to worry about API semantics like this.
I'm willing to negotiate if anyone else has different thoughts.

Reply:
Good idea, I'll change it, although I don't think it makes sense to try to address that for all endpoints in this PR, so I'll just make one for `assets` if that sounds good.

Reply:
That's fine, I just wanted to justify my philosophy.

Reply:
b951680